### PR TITLE
allow users to set cache_directory via env-var,  #1965

### PR DESCRIPTION
**Pull request by woutervh**
implements https://github.com/davidhalter/jedi/issues/1965

### CODE REVIEW

File: jedi/settings.py [hunk 1]
```diff
@@ -83,10 +85,17 @@
 """
 The path where the cache is stored.
 
-On Linux, this defaults to ``~/.cache/jedi/``, on OS X to
-``~/Library/Caches/Jedi/`` and on Windows to ``%LOCALAPPDATA%\\Jedi\\Jedi\\``.
+The cache-directory can be explicitly set by an environment-variable
+``JEDI_CACHE_DIR``.
+
+If this env-var is not set, then
```
Comment (davidhalter) [inline jedi/settings.py:91]: Can you revert this sentence? I don't like the grammar.

Comment (woutervh) [reply jedi/settings.py]: reverted.

**Comment by davidhalter** [comment]
Now it's not documented anymore. Sorry I don't have the patience for this. I initially rejected this, so I'm going to close.

**Comment by woutervh** [comment]
what does the word "revert" mean in your dictionary?

**Comment by woutervh** [comment]
Python applications and certainly libraries should never write anything to hardcoded unconfigurable directories.

I don't understand your objection to such no brainer.

**Pull request closed without merging**